src/plugins/fcm.ts

```diff
@@ -0,0 +1,98 @@
+import { Plugin, Cordova } from './plugin';
+import { Observable } from 'rxjs/Observable';
+
+
+/**
+ * @name FCM
+ * @description
+ * Push notification plugin for Cordova applications with Google Firebase FCM.
+ *
+ * @usage
+ * ```
+ * import {FCM} from 'ionic-native';
+ *
+ * // Listen for device token changes.
+ * FCM.onTokenRefresh()
+ *     .subscribe((token) => {
+ *         // Save/Update the token on the app server.
+ *     });
+ *
+ * // Define your application behaviour based on the notification data.
+ * FCM.onNotification()
+ *     .subscribe((data) => {
+ *         if (data.wasTapped) {
+ *             // Notification was received on device tray and tapped by the user.
+ *             alert(JSON.stringify(data));
+ *         } else {
+ *             // Notification was received in foreground, maybe the user needs to be notified.
+ *             alert(JSON.stringify(data));
+ *         }
+ *     });
+ * ```
+ */
+@Plugin({
+  pluginName: 'FCM',
+  plugin: 'cordova-plugin-fcm',
+  pluginRef: 'FCMPlugin',
+  repo: 'https://github.com/fechanique/cordova-plugin-fcm',
+  platforms: [
+    'Android',
+    'iOS'
+  ]
+})
+export class FCM {
+  /**
+   * Get notified every time a new token is generated.
+   * @return {Observable<any>}
+   */
+  @Cordova({
+    callbackStyle: 'node',
```

Comment:
`callbackStyle: 'node'` is used incorrectly here. It's not needed.

`callbackStyle: 'node'` is used when the plugin expects a callback like this one:
```
function(err, result) { }
```